### PR TITLE
Added promise returning for ios functions

### LINKED ISSUE #115: Add promise after finish save the file
Hi man!
If I try to send the file to my server automatically I stop recording, the file size is `0`.
I have to wait for ~1 second, so the file finished to be saved in the `audioPath`, and then send it to my server. And in this case, it works.

```javascript
let audio = {
    uri: audioPath,
    type: 'audio/aac',
    name: 'audio.aac'
};

let body = new FormData();
body.append('audio', audio);

xhr.open('POST', 'http://xxxx.com/xxx.php');
xhr.send(body);
```

If I execute that code exactly after calling `AudioRecorder.stopRecording();`, It will not work.
I have to put that code in a function and call it using `setTimeout(xxx, 2000);`

Maybe, the `AudioRecorder.stopRecording()` function should return a promise, so I can do 
```javascript
AudioRecorder.stopRecording().then(function(){
    // here send the tile to my server
});
```

If you have no time to do it, let me know and I can try to add this behavior and send a PR.

### PR DESCRIPTION
**Pull request by rakannimer**
Fixes #115

### CODE REVIEW

File: ios/AudioRecorderManager.m [hunk 1]
```diff
@@ -175,34 +177,38 @@ - (NSString *) applicationDocumentsDirectory
 
   if (error) {
       NSLog(@"error: %@", [error localizedDescription]);
-      // TODO: dispatch error over the bridge
+      reject(@"error: %@", [error localizedDescription]);
     } else {
       [_audioRecorder prepareToRecord];
+    resolve([_audioFileURL path]);
```
Comment (cooperka) [inline ios/AudioRecorderManager.m]: Nit: Fix indentation above

File: ios/AudioRecorderManager.m [hunk 2]
```diff
@@ -175,34 +177,38 @@ - (NSString *) applicationDocumentsDirectory
...
+  resolve([_audioFileURL path]);
 }
 
-RCT_EXPORT_METHOD(pauseRecording)
+RCT_EXPORT_METHOD(pauseRecording:(RCTPromiseResolveBlock)resolve reject:(__unused RCTPromiseRejectBlock)reject)
 {
   if (_audioRecorder.recording) {
     [self stopProgressTimer];
     [_audioRecorder pause];
+      resolve([_audioFileURL path]);
```
Comment (cooperka) [inline ios/AudioRecorderManager.m]: Nit: Fix indentation

**Comment by cooperka** [comment]
Code LGTM besides nits, thanks for doing this @RakanNimer. I can't test on a real device right now so I will not merge yet.

**Comment by RakanNimer** [comment]
Thanks for catching that, my bad, I missed it :D

**Comment by gcottrell1** [comment]
any chance of this getting merged? it would be very useful

**Comment by jsierles** [comment]
@RakanNimer Can you fix conflicts for a merge? Thanks!

**Comment by jsierles** [comment]
Closing this due to inactivity.

**Pull request closed without merging**